src/OpenSage.Game/Client/Drawable.cs

```diff
@@ -674,23 +1797,22 @@ public void Persist(StatePersister reader)
 
         reader.EndArray();
 
-        var unknownBool3 = true;
-        reader.PersistBoolean(ref unknownBool3);
-        if (!unknownBool3)
+        reader.PersistBoolean(ref _ambientSoundEnabled);
+
+        if (version >= 6)
         {
-            throw new InvalidStateException();
+            reader.PersistBoolean(ref _ambientSoundEnabledFromScript);
         }
 
         if (version >= 7)
         {
-            var unknownBool4 = true;
-            reader.PersistBoolean(ref unknownBool4);
-            if (!unknownBool4)
+            var customized = false;
+            reader.PersistBoolean(ref customized);
+
+            if (customized)
             {
-                throw new InvalidStateException();
+                throw new NotImplementedException();
```

Comment:
does this need a todo port?

Reply:
Good catch - cf776357ec7d2ca4572d63f9be9dcdeab330d946